### PR TITLE
Interproc_com

### PR DESCRIPTION
**Pull request by arafaarafa**

### CODE REVIEW

File: S_RTOS_02InterprocessCommunication/Task_2/Code/main.c [hunk 1]
```diff
@@ -97,12 +105,46 @@ static void prvSetupHardware( void );
...
 {
+	/*global vars*/
+	serial_mutex =  xSemaphoreCreateMutex();
+	
+	serial_task_1_cfg.data = (char *)"hello_from_serial_1";
+	serial_task_1_cfg.delay = 100;
+	serial_task_1_cfg.mutex = &serial_mutex;
+	
+	
+	serial_task_2_cfg.data = (char *)"hello_from_serial_2";
```
Comment (ahmedatef23) [inline S_RTOS_02InterprocessCommunication/Task_2/Code/main.c:116]: task 2 should have empty loop

File: S_RTOS_02InterprocessCommunication/Task_2/Code/main.c [hunk 2]
```diff
@@ -97,12 +105,46 @@ static void prvSetupHardware( void );
  */
 int main( void )
 {
+	/*global vars*/
+	serial_mutex =  xSemaphoreCreateMutex();
+	
+	serial_task_1_cfg.data = (char *)"hello_from_serial_1";
+	serial_task_1_cfg.delay = 100;
```
Comment (ahmedatef23) [inline S_RTOS_02InterprocessCommunication/Task_2/Code/main.c:112]: don't use magic number

File: S_RTOS_02InterprocessCommunication/Task_3/Code/main.c [hunk 3]
```diff
@@ -97,12 +112,77 @@ static void prvSetupHardware( void );
...
+	
+	
+	/*****************button cfg 1**********************/
+	button_task_2_cfg.id = BUTTON_TASK_2;
+	button_task_2_cfg.pin_num = PIN7;
+	button_task_2_cfg.queue = &xStructQueue;
+	
+	/*************consumer cfg**************/
+	consumer_task_cfg.id = CONSUMER_TASK;
+	consumer_task_cfg.delay = 100;
```
Comment (ahmedatef23) [inline S_RTOS_02InterprocessCommunication/Task_3/Code/main.c:137]: don't use magic number

Comment (arafaarafa) [reply S_RTOS_02InterprocessCommunication/Task_3/Code/main.c:137]: ok, I appreciate your comment, I will solve it.

File: S_RTOS_02InterprocessCommunication/Task_2/Code/main.c (hunk 1 above)
Comment (arafaarafa) [reply S_RTOS_02InterprocessCommunication/Task_2/Code/main.c:116]: you talk about any loop shall I have

File: S_RTOS_02InterprocessCommunication/Task_2/Code/main.c (hunk 2 above)
Comment (arafaarafa) [reply S_RTOS_02InterprocessCommunication/Task_2/Code/main.c:112]: https://github.com/arafaarafa/Sprints_Tasks/pull/9/files
solved all magic numbers for this pull

File: S_RTOS_02InterprocessCommunication/Task_2/Code/main.c (hunk 1 above)
Comment (ahmedatef23) [reply S_RTOS_02InterprocessCommunication/Task_2/Code/main.c:116]: the task shall have an empty loop that loops 100,000 times

File: S_RTOS_02InterprocessCommunication/Task_2/Code/serial_task/serial_task.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+void serial_task(void *pvParameters){
+	uint32_t load_counter;
+	uint8_t msg_cnt;
+	for( ; ; ){
+			if(xSemaphoreTake( *(((serial_task_config *)pvParameters)->mutex),  ( TickType_t ) 100) == pdTRUE){
+				for(msg_cnt=0; msg_cnt<10; msg_cnt++){
+					vSerialPutString((const signed char*)(((serial_task_config *)pvParameters)->data), 20);
+					if(((serial_task_config *)pvParameters)->delay>100){
+						for(load_counter=0; load_counter< 100000; load_counter++);
```
Comment (ahmedatef23) [inline S_RTOS_02InterprocessCommunication/Task_2/Code/serial_task/serial_task.c:12]: is that mean every task you create has a 100000 loop?

**Pull request merged**